connect/mirror/src/test/java/org/apache/kafka/connect/mirror/integration/MirrorConnectorsWithCustomForwardingAdminIntegrationTest.java

```diff
@@ -0,0 +1,553 @@
+import org.apache.kafka.common.acl.AclOperation;
+import org.apache.kafka.common.acl.AclPermissionType;
+import org.apache.kafka.common.config.ConfigResource;
+import org.apache.kafka.common.config.TopicConfig;
+import org.apache.kafka.common.resource.PatternType;
+import org.apache.kafka.common.resource.ResourcePattern;
+import org.apache.kafka.common.resource.ResourcePatternFilter;
+import org.apache.kafka.common.resource.ResourceType;
+import org.apache.kafka.common.utils.Exit;
+import org.apache.kafka.common.utils.Utils;
+import org.apache.kafka.connect.connector.Connector;
+import org.apache.kafka.connect.mirror.DefaultConfigPropertyFilter;
+import org.apache.kafka.connect.mirror.MirrorCheckpointConnector;
+import org.apache.kafka.connect.mirror.MirrorHeartbeatConnector;
+import org.apache.kafka.connect.mirror.MirrorMakerConfig;
+import org.apache.kafka.connect.mirror.MirrorSourceConnector;
+import org.apache.kafka.connect.mirror.SourceAndTarget;
+import org.apache.kafka.connect.mirror.clients.admin.FakeForwardingAdminWithLocalMetadata;
+import org.apache.kafka.connect.mirror.clients.admin.FakeLocalMetadataStore;
+import org.apache.kafka.connect.util.clusters.EmbeddedConnectCluster;
+import org.apache.kafka.connect.util.clusters.EmbeddedKafkaCluster;
+import org.apache.kafka.connect.util.clusters.UngracefulShutdownException;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+import static org.apache.kafka.connect.mirror.MirrorConnectorConfig.FORWARDING_ADMIN_CLASS;
+import static org.apache.kafka.connect.mirror.TestUtils.generateRecords;
+import static org.apache.kafka.connect.mirror.integration.MirrorConnectorsIntegrationBaseTest.basicMM2Config;
+import static org.apache.kafka.connect.mirror.integration.MirrorConnectorsIntegrationBaseTest.waitUntilMirrorMakerIsRunning;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.CONNECTOR_CLIENT_ADMIN_OVERRIDES_PREFIX;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.CONNECTOR_CLIENT_CONSUMER_OVERRIDES_PREFIX;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.CONNECTOR_CLIENT_PRODUCER_OVERRIDES_PREFIX;
+import static org.apache.kafka.test.TestUtils.waitForCondition;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Tests MM2 is using provided ForwardingAdmin to create/alter topics, partitions and ACLs.
+ */
+@Tag("integration")
+public class MirrorConnectorsWithCustomForwardingAdminIntegrationTest {
```

Comment:
There's a lot of boilerplate logic, have you considered extending `MirrorConnectorsIntegrationBaseTest`?

This will require some changes in `MirrorConnectorsIntegrationBaseTest` but I wonder if it's worth it.

Reply:
Setting up `EmbededCluster` with `AclAuthorizer` to test ACLs sync was a lot of work, this is why I separated it away from `MirrorConnectorsIntegrationBaseTest`. I can give it another try and see how much work is needed to move some of this to `MirrorConnectorsIntegrationBaseTest`.

Reply:
Yes it will require changes in `EmbededCluster` too. I'm ok merging as is and maybe we can follow up in a separate PR?

Reply:
Yeah, I think this will be better. I can open a Jira and raise another PR to have an easier way to set up `EmbeddedConnectCluster` and `EmbeddedKafkaCluster` with `AclAuthorizer` to clean this up in MM2 integration tests and `ExactlyOnceSourceIntegrationTest` in connect as well.

Reply:
@mimaison I created this jira [https://issues.apache.org/jira/browse/KAFKA-14344](https://issues.apache.org/jira/browse/KAFKA-14344) for `EmbeddedKafkaCluster`. The easiest solution is adding a new parameter `additionalClientConfigs` to `EmbeddedKafkaCluster`  constructor. I raised a pr [https://github.com/apache/kafka/pull/12804](https://github.com/apache/kafka/pull/12804)  and then I can fix this pr to use `MirrorConnectorsIntegrationBaseTest` once pr#12804 gets merged.

Reply:
Do you prefer to get #12804 merged first so you can update this PR?

Reply:
yes this would be better than raising another pr to use `MirrorConnectorsIntegrationBaseTest` for the pr test.

Reply:
The test now extends `MirrorConnectorsIntegrationBaseTest`. 
I also fixed the following tests
- `IdentityReplicationIntegrationTest.testOneWayReplicationWithAutoOffsetSync` 
- `MirrorConnectorsIntegrationBaseTest.testOneWayReplicationWithAutoOffsetSync`
Both test cases are supposed to test that MirrorMaker is replicating in one direction only however the assertions were testing that topics are created on both clusters.